src/libsystemd/sd-event/sd-event.c

```diff
@@ -4000,8 +4000,17 @@ static int source_memory_pressure_write(sd_event_source *s) {
         if (s->memory_pressure.write_buffer_size > 0) {
                 n = write(s->memory_pressure.fd, s->memory_pressure.write_buffer, s->memory_pressure.write_buffer_size);
                 if (n < 0) {
-                        if (!ERRNO_IS_TRANSIENT(errno))
-                                return -errno;
+                        if (!ERRNO_IS_TRANSIENT(errno)) {
+                                /* If kernel is built with CONFIG_PSI_DEFAULT_DISABLED it will expose PSI
+                                 * files, but then generates EOPNOSUPP on read() and write() (instead of on
```

Comment:
shouldn't this check for EOPNOTSUPP then?

Reply:
I don't think so, no.

My thinking on this is that PSI information is useful but not essential. As the issue this addresses shows there are unexpected problems with the PSI interface with some conditions, and previously if we noticed any we'd completely fail and terminate the program. But that' just too radical for some mostly auxiliary part of the programs, where the programs can just continue running pretty much just fine.

So this is definitely worth to log about, but I am pretty sure it's never a reason to terminate the program. Hence I just added graceful handling for any kind of error here, not just for the one we actually saw IRL.

does that make sense?